app/controllers/system_notifications_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class SystemNotificationsController < ApplicationController
+  def index
+    @system_notifications = prepare_notifications
```

Comment:
Call `prepare_notifications` on `index` with `before_action` instead, and set `@system_notifications` in there.